app/connectors/GetBusinessDetailsConnector.scala

```diff
@@ -54,6 +54,9 @@ class GetBusinessDetailsConnector @Inject()(val http: HttpClient,
                 valid
               }
             )
+          case NOT_FOUND =>
+            logger.warn(s"[GetBusinessDetailsConnector][getBusinessDetails] - RESPONSE status: ${response.status}, body: ${response.body}")
+            IncomeSourceDetailsError(response.status, response.body)
```

Comment:
It will be more appropriate to introduce something like: IncomeSourceDetailsNotFound (extend IncomeSourceDetailsResponseModel)

Reply:
Done